server/src/main/java/com/hedera/block/server/persistence/storage/read/BlockAsDirReader.java

```diff
@@ -144,6 +151,9 @@ private Optional<BlockItem> readBlockItem(@NonNull final String blockItemPath)
             // FileNotFound is also thrown when a file cannot be read.
             // So re-throw here to make a different decision upstream.
             throw io;
+        } catch (ParseException e) {
+            LOGGER.log(System.Logger.Level.ERROR, "Error parsing block item: " + blockItemPath, e);
+            throw new IOException(e);
```

Comment:
Wrapping this in IOException is not good, it effectively _hides_ the real failure.
Much better to declare the `ParseException` and handle that further up (when we can return a better failure code or take some other reasonable action).

Reply:
Fixed